pkg/controller/volume/scheduling/scheduler_binder_test.go

```diff
@@ -2002,7 +2012,7 @@ func TestBindPodVolumes(t *testing.T) {
 		claimsToProvision := []*v1.PersistentVolumeClaim{}
 		if !scenario.bindingsNil {
 			if scenario.binding != nil {
-				bindings = []*BindingInfo{scenario.binding}
+				bindings = []*BindingInfo{makeBinding(scenario.binding.pvc, scenario.binding.pv)}
```

Comment:
Why do you call `makeBinding` twice? In scenario definitions and here.

Reply:
If don't add this, it will fail when run stress.
``` log
5s: 0 runs so far, 0 failures
10s: 0 runs so far, 0 failures
15s: 0 runs so far, 0 failures
20s: 0 runs so far, 0 failures

/var/folders/6v/7stmg2756wlfk9c_qnv1hnbm0000gn/T/go-stress-20210227T142520-109974642
==================
WARNING: DATA RACE
Read at 0x00c000424cc8 by goroutine 223:
  k8s.io/kubernetes/pkg/controller/volume/scheduling.(*testEnv).assumeVolumes()
      /Users/zsm/go/src/k8s.io/kubernetes/pkg/controller/volume/scheduling/scheduler_binder_test.go:401 +0x11a
  k8s.io/kubernetes/pkg/controller/volume/scheduling.TestBindPodVolumes.func6()
      /Users/zsm/go/src/k8s.io/kubernetes/pkg/controller/volume/scheduling/scheduler_binder_test.go:2023 +0xbaa
  k8s.io/kubernetes/pkg/controller/volume/scheduling.TestBindPodVolumes.func7()
      /Users/zsm/go/src/k8s.io/kubernetes/pkg/controller/volume/scheduling/scheduler_binder_test.go:2068 +0x101
  testing.tRunner()
      /usr/local/Cellar/go/1.15.6/libexec/src/testing/testing.go:1123 +0x202

Previous write at 0x00c000424cc8 by goroutine 81:
  k8s.io/kubernetes/pkg/controller/volume/scheduling.(*volumeBinder).bindAPIUpdate()
      /Users/zsm/go/src/k8s.io/kubernetes/pkg/controller/volume/scheduling/scheduler_binder.go:507 +0x8c5
  k8s.io/kubernetes/pkg/controller/volume/scheduling.(*volumeBinder).BindPodVolumes()
      /Users/zsm/go/src/k8s.io/kubernetes/pkg/controller/volume/scheduling/scheduler_binder.go:442 +0x376
  k8s.io/kubernetes/pkg/controller/volume/scheduling.TestBindPodVolumes.func6()
      /Users/zsm/go/src/k8s.io/kubernetes/pkg/controller/volume/scheduling/scheduler_binder_test.go:2054 +0x301
  k8s.io/kubernetes/pkg/controller/volume/scheduling.TestBindPodVolumes.func7()
      /Users/zsm/go/src/k8s.io/kubernetes/pkg/controller/volume/scheduling/scheduler_binder_test.go:2068 +0x101
  testing.tRunner()
      /usr/local/Cellar/go/1.15.6/libexec/src/testing/testing.go:1123 +0x202

Goroutine 223 (running) created at:
  testing.(*T).Run()
      /usr/local/Cellar/go/1.15.6/libexec/src/testing/testing.go:1168 +0x5bb
  k8s.io/kubernetes/pkg/controller/volume/scheduling.TestBindPodVolumes()
      /Users/zsm/go/src/k8s.io/kubernetes/pkg/controller/volume/scheduling/scheduler_binder_test.go:2066 +0x19d3
  testing.tRunner()
      /usr/local/Cellar/go/1.15.6/libexe
…
25s: 3 runs so far, 1 failures (33.33%)
```

Reply:
`scenario.binding` should be local to each test case. If not, that seems to indicate there is still some shared state somewhere.